switchbot_api/__init__.py

```diff
@@ -189,12 +202,21 @@ async def _request(self, path: str = "", callback: str = "get", json=None):
             ) as response:
                 if response.status == 403:
                     raise InvalidAuth()
+
                 body = await response.json()
-                if response.status >= 400 or body.get("statusCode") != 100:
-                    _LOGGER.error("Error %s: %s", response.status, body)
-                    raise CannotConnect()
-                return body.get("body")
 
+                if response.status >= 400:
+                    raise CannotConnect()
+                    
+                match body.get("statusCode"):
+                    case 100:
+                        return body.get("body")
+                    case 171:
```

Comment:
I think for the device, it should be 161, 171 is for an offline hub.

Reply:
Hmm, in my testing my Plugs definitely throw 171 when they're disconnected from power. 

I'll try with my Ceiling Light Pro to see what that throws up...

Reply:
Both the Ceiling Light, and the virtual remote devices also return 171 when offline. I've added 161 to match on both and a comment explaining it :)

Reply:
Oh I see, thanks for handling both cases and adding a comment explaining the situation :)